server/middleware/registration/index.js

```diff
@@ -0,0 +1,126 @@
+const bcrypt = require('bcrypt');
+const {
+  HttpStatusCodes: { StatusCodes },
+  DataException
+} = require('../../utils/constants');
+const { REQUEST_TIMEOUT, OK, BAD_REQUEST, GATEWAY_TIMEOUT } = StatusCodes;
+const mongoUtil = require('../../utils/mongoUtil');
+const JWT = require('jsonwebtoken');
+const { v4 } = require('uuid');
+const registrationLogger = require('./registrationLogger');
+
+const resObj = (statusCode, message, data = {}) => ({
+  statusCode,
+  message,
+  data,
+});
+
+const isFormValid = ({ body }) => {
+  const formFields = [
+    // 'role_id',
+    'firstName',
+    'lastName',
+    'title',
+    'email',
+    'password'
+  ];
+  
+  for (let i = 0; i < formFields.length; i++) {
+    const field = formFields[i];
+    if (!Object.hasOwnProperty.call(body, formFields[i]) || body[field].length === 0) {
+      return false;
+    }
+  }
+
+  const emailPattern = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  if (!emailPattern.test(body.email)) {
+    return false;
+  } else {
+    return true;
+  }
+
+};
+
+const hashForm = ({ body }) => {
+  const saltRounds = 10;
+  body.email = body.email.toLowerCase();
+  const { password, ...rest } = body;
+  const userPayload = { password: bcrypt.hashSync(password, saltRounds), ...rest };
+  return userPayload;
+};
+
+const generateCookie = (res, userPayload) => {
+  const { DOMAIN, SECRET_KEY } = process.env;
+  const { password, ...rest } = userPayload;
+  const cookieExp = new Date(Date.now() + 8 * 3600000);
+  const options = {
+    expires: cookieExp,
+    path: '/',
+    domain: DOMAIN || 'femmecubator.com',
+  };
+  const token = JWT.sign(rest, SECRET_KEY);
```

Comment:
we want to pick what data we're going to include in the token; also, there were some change requirements that will affect some UI components so we have to generate userName attrib that we used to get from the form data before, in this case, we would have to join the firstName and the first character of the last name, for example: "Jackson C." See the code snippet below:

`const { email, role_id, firstName, lastName } = userPayload;`
`...etc ...etc`
`const token = JWT.sign({ email, role_id, userName: `\``${firstName} ${lastName.charAt(0)}.`\`` }, SECRET_KEY);`